include/z64collision_check.h

```diff
@@ -9,90 +9,140 @@
-    /* 0x07 */ u8 shape;   // JntSph, Cylinder, Tris, or Quad
+    /* 0x04 */ u8 atFlags;
+    /* 0x05 */ u8 acFlags;
+    /* 0x06 */ u8 ocFlags1;
+    /* 0x07 */ u8 shape;
 } ColliderInitToActor; // size = 0x08
 
 typedef struct {
     /* 0x00 */ u32 dmgFlags; // Toucher damage type flags.
     /* 0x04 */ u8 effect; // Damage Effect (Knockback, Fire, etc.)
     /* 0x05 */ u8 damage; // Damage or Stun Timer
-} ColliderTouch; // size = 0x08
+} ColliderElementTouch; // size = 0x08
 
 typedef struct {
     /* 0x00 */ u32 dmgFlags;  // Bumper damage type flags.
     /* 0x04 */ u8 effect;  // Damage Effect (Knockback, Fire, etc.)
     /* 0x05 */ u8 defense; // Damage Resistance
     /* 0x06 */ Vec3s hitPos; // Point of contact
-} ColliderBump; // size = 0x0C
+} ColliderElementBump; // size = 0x0C
 
 typedef struct {
     /* 0x00 */ u32 dmgFlags; // Bumper exclusion mask
     /* 0x04 */ u8 effect; // Damage Effect (Knockback, Fire, etc.)
     /* 0x05 */ u8 defense; // Damage Resistance
-} ColliderBumpInit; // size = 0x08
+} ColliderElementBumpInit; // size = 0x08
+
+/**
+ * Affects the sound Link's sword makes when hitting it, hookability,
+ * and possibly other things. It's definitely not flags, as all checks
+ * are == or !=. Will probably need more actors decomped to truly
+ * understand what this is.
+ */
+typedef enum {
+    /* 0 */ ELEMTYPE_UNK0,
+    /* 1 */ ELEMTYPE_UNK1,
+    /* 2 */ ELEMTYPE_UNK2,
+    /* 3 */ ELEMTYPE_UNK3,
+    /* 4 */ ELEMTYPE_UNK4,
+    /* 5 */ ELEMTYPE_UNK5,
+    /* 6 */ ELEMTYPE_UNK6,
+    /* 7 */ ELEMTYPE_UNK7
+} ElementType;
 
-typedef struct ColliderInfo {
-    /* 0x00 */ ColliderTouch toucher; // Damage properties when acting as an AT collider
-    /* 0x08 */ ColliderBump bumper; // Damage properties when acting as an AC collider
+typedef struct ColliderElement {
+    /* 0x00 */ ColliderElementTouch toucher; // Damage properties when acting as an AT collider
+    /* 0x08 */ ColliderElementBump bumper; // Damage properties when acting as an AC collider
     /* 0x14 */ u8 elemType; // Affects sfx reaction when attacked by Link and hookability. Full purpose unknown.
     /* 0x15 */ u8 toucherFlags; // Information flags for AT collisions
     /* 0x16 */ u8 bumperFlags; // Information flags for AC collisions
     /* 0x17 */ u8 ocElemFlags; // Information flags for OC collisions
     /* 0x18 */ Collider* atHit;                // object touching this element's AT collider
     /* 0x1C */ Collider* acHit;                // object touching this element's AC collider
-    /* 0x20 */ struct ColliderInfo* atHitInfo; // element that hit the AT collider
-    /* 0x24 */ struct ColliderInfo* acHitInfo; // element that hit the AC collider
-} ColliderInfo; // size = 0x28
+    /* 0x20 */ struct ColliderElement* atHitInfo; // element that hit the AT collider
+    /* 0x24 */ struct ColliderElement* acHitInfo; // element that hit the AC collider
+} ColliderElement; // size = 0x28
```

Comment:
This seems like an excellent way to mix up the general collider properties with those of the sub-collider things that JntSphs have.

Reply:
Can you elaborate? `ColliderElement` is exactly the base struct for `ColliderJntSphElement`

Reply:
Oh, I thought it lived in the collider itself like it does for the other types.

Reply:
I can't fix illiteracy <img src="https://cdn.frankerfacez.com/emoticon/263833/2" height="20px">

Reply:
There is the issue that the base struct for JntSph just behaves completely differently, it feels like there ought to be some sort of distinction.

Reply:
wdym it behaves completely differently

For example if you look at an AT Cylinder vs an AC JntSph

`CollisionCheck_AT` checks the cylinder's `Collider.atFlags`
then `CollisionCheck_AC` checks the jntSph's `Collider.acFlags`, (and the `AC_TYPE_ALL` bits between cylinder's `Collider.atFlags` and jntSph's `Collider.acFlags`)
then `CollisionCheck_ATCylVsACJntSph` makes use of `ColliderElement atCyl->elem` and `ColliderElement acJntSph->elements[...].base`

outside VS funcs: use `Collider` base
inside VS funcs: use `ColliderElement` base (or elem for things like cylinder with a single element)

Reply:
Also note the symmetry in `CollisionCheck_SetATvsAC` with how it handles all shapes the same

Reply:
Can this be considered resolve?
I'll consider so in a week or so